src/main/java/com/cloudbees/jenkins/support/filter/DefaultContentFilter.java

```diff
@@ -0,0 +1,277 @@
+
+    public DefaultContentFilter() {
+        stopWords.addAll(Arrays.asList(
+                "jenkins", "node", "master", "computer",
+                "item", "label", "view", "all", "unknown",
+                "user", "anonymous", "authenticated"
+        ));
+        load();
+    }
+
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    @DataBoundSetter
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        save();
+    }
+
+    public Map<String, String> getMappings() {
+        Map<String, String> map = new HashMap<>(mappings.size());
+        for (Map.Entry<String, Replacer> entry : mappings.entrySet()) {
+            map.put(entry.getKey(), entry.getValue().replacement);
+        }
+        return map;
+    }
+
+    public Set<String> getStopWords() {
+        return stopWords;
+    }
+
+    @RequirePOST
+    public void doUpdate(StaplerRequest req, StaplerResponse res) throws ServletException, IOException {
+        req.bindJSON(this, req.getSubmittedForm());
+        res.forwardToPreviousPage(req);
+    }
+
+    @Override
+    public @Nonnull String filter(@Nonnull String input) {
+        if (input.isEmpty() || !enabled) return input;
+        String filtered = input;
+        for (Replacer replacer : mappings.values()) {
+            filtered = replacer.replaceAll(filtered);
+        }
+        return filtered;
+    }
+
+    private XmlFile getConfigFile() {
+        return new XmlFile(new File(Jenkins.get().getRootDir(), "secrets/" + getClass().getCanonicalName() + ".xml"));
+    }
+
+    @Override
+    public synchronized void reload() {
+        ExtensionList.lookup(AbstractItem.class).forEach(item -> {
+            stopWords.add(item.getTaskNoun());
+            stopWords.add(item.getPronoun());
+        });
+        SensitiveNameProvider.all().stream()
+                .flatMap(provider -> provider.names().map(name -> ImmutablePair.of(name, provider.prefix())))
+                .filter(pair -> !stopWords.contains(pair.left.toLowerCase(Locale.ENGLISH)))
+                .forEach(pair -> mappings.computeIfAbsent(pair.left, ignored -> Replacer.of(pair.left, pair.right)));
+    }
+
+    @Override
+    public synchronized void save() {
+        reload();
+        XmlFile file = getConfigFile();
+        SerializationProxy proxy = new SerializationProxy();
+        proxy.setEnabled(enabled);
+        Map<String, String> mappings = new HashMap<>(this.mappings.size());
+        for (Map.Entry<String, Replacer> entry : this.mappings.entrySet()) {
+            mappings.put(entry.getKey(), entry.getValue().replacement);
+        }
+        proxy.setMappings(mappings);
+        proxy.setStopWords(new HashSet<>(stopWords));
+        try {
+            file.write(proxy);
+        } catch (IOException e) {
+            logWarning(e, "Could not save file %s", file.getFile().getName());
+        }
+    }
+
+    private synchronized void load() {
```

Comment:
it is only called from the constructor and is private, Should it be synchronized?

Reply:
I think a better question I had here was whether or not I should make this method public. It was public at one point, but I minimized access once I wasn't using it outside the constructor. The synchronization would be technically unnecessary in the constructor though, indeed.